packages/plugin-ext/src/main/browser/plugin-ext-frontend-module.ts

```diff
@@ -70,20 +65,97 @@ import { PluginCommandOpenHandler } from './plugin-command-open-handler';
+    bind(ConnectionMainImpl).toSelf().inTransientScope();
+    bind(TasksMainImpl).toSelf().inTransientScope();
+    bind(LanguagesContributionMainImpl).toSelf().inTransientScope();
+    bind(DebugMainImpl).toSelf().inTransientScope();
+    bind(ScmMainImpl).toSelf().inTransientScope();
+    bind(DecorationsMainImpl).toSelf().inTransientScope();
+    bind(WindowStateMain).toSelf().inTransientScope();
+    bind(ClipboardMainImpl).toSelf().inTransientScope();
+    bind(TextEditorsMainImpl).toSelf().inTransientScope();
+    bind(DocumentsMainImpl).toSelf().inTransientScope();
+
+    bindContributionProvider(bind, RPCProtocolServiceProvider);
+    bind(RPCProtocolServiceProvider).to(CommandRegistryMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(QuickOpenMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(LanguagesMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(OutputChannelRegistryMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(WorkspaceMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(DialogsMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(MessageRegistryMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(PreferenceRegistryMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(StatusBarMessageRegistryMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(EnvMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(NotificationMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(TerminalMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(TreeViewsMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(WebviewsMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(PluginStorageMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(ConnectionMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(TasksMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(LanguagesContributionServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(DebugMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(ScmMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(DecorationsMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(WindowStateMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(ClipboardMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(DocumentsMainServiceProvider).inSingletonScope();
+    bind(RPCProtocolServiceProvider).to(TextEditorsMainServiceProvider).inSingletonScope();
```

Comment:
providers should go into transient scope as well, otherwise they cache instances of main services